fuel-core/src/schema/dap.rs

```diff
@@ -192,4 +192,272 @@ impl DapMutation {
+            .ok_or_else(|| async_graphql::Error::new("VM not found"))?;
+
+        vm.set_single_stepping(enable);
+        Ok(enable)
+    }
+
+    #[cfg(not(feature = "debug"))]
+    async fn set_breakpoint(
+        &self,
+        _ctx: &Context<'_>,
+        _id: ID,
+        _breakpoint: self::gql_types::Breakpoint,
+    ) -> async_graphql::Result<bool> {
+        Err(async_graphql::Error::new(
+            "Feature 'debug' is not compiled in",
+        ))
+    }
+
+    #[cfg(feature = "debug")]
+    async fn set_breakpoint(
+        &self,
+        ctx: &Context<'_>,
+        id: ID,
+        breakpoint: self::gql_types::Breakpoint,
+    ) -> async_graphql::Result<bool> {
+        trace!("Continue exectuon of VM {:?}", id);
+
+        let mut locked = ctx.data_unchecked::<GraphStorage>().lock().await;
+        let vm = locked
+            .vm
+            .get_mut(&id)
+            .ok_or_else(|| async_graphql::Error::new("VM not found"))?;
+
+        vm.set_breakpoint(breakpoint.into());
+        Ok(true)
+    }
+
+    async fn start_tx(
+        &self,
+        ctx: &Context<'_>,
+        id: ID,
+        tx_json: String,
+    ) -> async_graphql::Result<self::gql_types::RunResult> {
+        trace!("Spawning a new VM instance");
+
+        dbg!(&tx_json);
+
+        let tx: Transaction = serde_json::from_str(&tx_json)
+            .map_err(|_| async_graphql::Error::new("Invalid transaction JSON"))?;
+
+        let mut locked = ctx.data_unchecked::<GraphStorage>().lock().await;
+        let vm = locked
+            .vm
+            .get_mut(&id)
+            .ok_or_else(|| async_graphql::Error::new("VM not found"))?;
+
+        let state_ref = vm
+            .transact(tx)
+            .map_err(|err| async_graphql::Error::new(format!("Fransaction failed: {err:?}")))?;
+
+        #[cfg(feature = "debug")]
+        {
+            let dbgref = state_ref.state().debug_ref();
+            Ok(self::gql_types::RunResult {
+                state: match dbgref {
+                    Some(_) => self::gql_types::RunState::Breakpoint,
+                    None => self::gql_types::RunState::Completed,
+                },
+                breakpoint: dbgref.and_then(|d| match d {
+                    DebugEval::Continue => None,
+                    DebugEval::Breakpoint(bp) => Some(bp.into()),
+                }),
+            })
+        }
+
+        #[cfg(not(feature = "debug"))]
+        {
+            let _ = state_ref;
+            Ok(self::gql_types::RunResult {
+                state: self::gql_types::RunState::Completed,
+                breakpoint: None,
+            })
+        }
+    }
+
+    #[cfg(not(feature = "debug"))]
+    async fn continue_tx(
+        &self,
+        _ctx: &Context<'_>,
+        _id: ID,
+    ) -> async_graphql::Result<self::gql_types::RunResult> {
+        Err(async_graphql::Error::new(
+            "Feature 'debug' is not compiled in",
+        ))
+    }
+
+    #[cfg(feature = "debug")]
+    async fn continue_tx(
```

Comment:
Do we really need the `_tx` suffix here? As long we provide a valid state id, it should be implicit

Reply:
`continue` is a reserved identifier, so it would be quite awkward to use.